permission-portal-frontend/client/src/screens/CodeValidations.js

```diff
@@ -8,70 +7,62 @@ import { withStore } from '../store'
 import { Redirect } from 'react-router-dom'
 import { observer } from 'mobx-react'
 import PageTitle from '../components/PageTitle'
+import PendingOperationButton from '../components/PendingOperationButton'
 
 // snackbars docs can be found here:
 // https://material-ui.com/components/snackbars/
 
 const CodeValidationsBase = observer((props) => {
-  const [code, setCode] = useState('')
-  const [date, setDate] = useState('')
+  const [code, setCode] = useState('123-45-6')
+  const [toastInfo, setToastInfo] = useState({
+    success: false,
+    msg: '',
+  })
 
   let confirmedToast = createRef()
 
+  //TODO show confired toast when code confirmed by app
+
+  const genNewCode = async () => {
+    // TODO get new code from server
+    try {
+      setCode('123-45-9')
+    } catch (err) {
+      setToastInfo({ success: false, msg: 'Could not generate new code, please try again' })
+      confirmedToast.current.show()
+    }
+  }
+
   return props.store.user.isSignedIn ? (
     <div className="module-container">
-      {/* this is a Snackbar template to use for the success/failure notifications */}
-
       <PageTitle title="Positive Test Validations" />
       <h1>Positive Test Validations</h1>
       <div id="actions-box" className="gray-background">
-        <div className="action-section">
+        <div className="validation-container">
           <div className="section-heading-container">
-            <h2 className="section-heading">Validation Code</h2>
+            <h2>Validation Code</h2>
             <div className="tooltip">
               <img src={info_icon} alt="info" />
               <div className="tooltip-msg">
-                {/* to replace tooltip text here: */}
-                <div className="tooltip-title">This the tooltip title</div>
+                {/* TODO replace tooltip text */}
+                <div className="tooltip-title">Test Validation Codes</div>
                 <div className="tooltip-body">
                   Quisque sagittis, vel hendrerit consectetur tincidunt senectus. Feugiat aenean nunc, tempus tempus,
```

Comment:
nit: I'd put a TODO in here so it is not missed (easier to search for things that are not done)

Reply:
I believe that's what I did with the changes at lines 47-48. But let me know if I misinterpreted what you meant.